build-tools/src/main/kotlin/org/jetbrains/kotlin/benchmark/SwiftBenchmarkingPlugin.kt

```diff
@@ -138,89 +81,16 @@ open class SwiftBenchmarkingPlugin: Plugin<Project> {
-                val nativeCompileTime = getNativeCompileTime(applicationName, benchmark.compileTasks)
-                val benchContents = buildDir.resolve(nativeBenchResults).readText()
-                val frameworkPath = framework.outputFile.absolutePath
-                val codeSizeEntity = if (benchmark.useCodeSize == CodeSizeEntity.FRAMEWORK)
-                    File("$frameworkPath/$NATIVE_FRAMEWORK_NAME").canonicalPath
-                else
-                    executable.toString()
-
-                val properties = commonBenchmarkProperties + mapOf(
-                    "type" to "native",
-                    "compilerVersion" to konanVersion,
-                    "flags" to nativeTarget.compilations.main.kotlinOptions.freeCompilerArgs.map { "\"$it\"" },
-                    "benchmarks" to benchContents,
-                    "compileTime" to listOf(nativeCompileTime),
-                    "codeSize" to getCodeSizeBenchmark(applicationName, codeSizeEntity)
-                )
-
-                val output = createJsonReport(properties)
-                buildDir.resolve(nativeJson).writeText(output)
-            }
-        }
-
-        // JVM run task.
-        val jvmRun = tasks.create("jvmRun", RunJvmTask::class.java) {
-            it.doLast {
-                println("JVM run is unsupported")
-            }
-        }
-
-        // JVM report task.
-        val jvmJsonReport = tasks.create("jvmJsonReport") {
-
-            it.group = BENCHMARKING_GROUP
-            it.description = "Builds the benchmarking report for Kotlin/JVM."
-
-            it.doLast {
-                println("JVM run is unsupported")
-            }
-
-            jvmRun.finalizedBy(it)
-        }
     }
 
-    override fun apply(target: Project) = with(target) {
-        pluginManager.apply("kotlin-multiplatform")
-
-        // Use Kotlin compiler version specified by the project property.
-        dependencies.add("kotlinCompilerClasspath", "org.jetbrains.kotlin:kotlin-compiler-embeddable:$kotlinVersion")
-
-        extensions.create(BENCHMARK_EXTENSION_NAME, SwiftBenchmarkExtension::class.java, this)
-        configureMPPExtension()
-        addTimeListener(this)
-        configureTasks()
-    }
-
-    companion object {
-        const val NATIVE_TARGET_NAME = "native"
-        const val NATIVE_FRAMEWORK_NAME = "benchmark"
-        const val BENCHMARK_EXTENSION_NAME = "swiftBenchmark"
+    override fun Project.collectCodeSize(applicationName: String) =
+            getCodeSizeBenchmark(applicationName,
+                    if (benchmark.useCodeSize == CodeSizeEntity.FRAMEWORK)
+                        File("${framework.outputFile.absolutePath}/$nativeFrameworkName").canonicalPath
+                    else
+                        nativeExecutable
+            )
 
-        const val BENCHMARKING_GROUP = "benchmarking"
-    }
+    override fun Project.getCompilerFlags(nativeTarget: KotlinNativeTarget) =
```

Comment:
Why this is empty for Swift benchmarks?

Reply:
We are interested only in flags that influence